src/pages/distroSettings/tabs/TaskTab/getFormSchema.ts

```diff
@@ -0,0 +1,183 @@
+                type: "number" as "number",
+                title: "Patch Factor (0 to 100 inclusive)",
+                default: 0,
+                minimum: 0,
+                maximum: 100,
+              },
+              patchTimeInQueueFactor: {
+                type: "number" as "number",
+                title: "Patch Time in Queue Factor (0 to 100 inclusive)",
+                default: 0,
+                minimum: 0,
+                maximum: 100,
+              },
+              mainlineTimeInQueueFactor: {
+                type: "number" as "number",
+                title: "Mainline Time in Queue Factor (0 to 100 inclusive)",
+                default: 0,
+                minimum: 0,
+                maximum: 100,
+              },
+              commitQueueFactor: {
+                type: "number" as "number",
+                title: "Commit Queue Factor (0 to 100 inclusive)",
+                default: 0,
+                minimum: 0,
+                maximum: 100,
+              },
+              expectedRuntimeFactor: {
+                type: "number" as "number",
+                title: "Expected Runtime Factor (0 to 100 inclusive)",
+                default: 0,
+                minimum: 0,
+                maximum: 100,
+              },
+              generateTaskFactor: {
+                type: "number" as "number",
+                title: "Generate Task Factor (0 to 100 inclusive)",
+                default: 0,
+                minimum: 0,
+                maximum: 100,
+              },
+              groupVersions: {
+                type: "boolean" as "boolean",
+                title: "Group versions",
+                default: false,
+              },
+            },
+          },
+        },
+      },
+      dispatcherSettings: {
+        type: "object" as "object",
+        title: "Task Dispatcher",
+        properties: {
+          version: {
+            type: "string" as "string",
+            title: "Task Dispatcher Version",
+            oneOf: [
+              {
+                type: "string" as "string",
+                title: "Revised",
+                enum: [DispatcherVersion.Revised],
+              },
+              {
+                type: "string" as "string",
+                title: "Revised with dependencies",
+                enum: [DispatcherVersion.RevisedWithDependencies],
+              },
+            ],
+          },
+        },
+      },
+    },
+  },
+  uiSchema: {
+    finderSettings: {
+      "ui:ObjectFieldTemplate": CardFieldTemplate,
+      version: {
+        "ui:allowDeselect": false,
+      },
+    },
+    plannerSettings: {
+      "ui:ObjectFieldTemplate": CardFieldTemplate,
+      version: {
+        "ui:allowDeselect": false,
+      },
+      tunableOptions: {
+        "ui:data-cy": "tunable-options",
+        // Options are only available if the planner version is tunable.
+        ...(plannerVersion === PlannerVersion.Legacy && {
+          "ui:widget": "hidden",
+        }),
```

Comment:
Any chance of using [`dependencies`](https://react-jsonschema-form.readthedocs.io/en/v4.2.2/usage/dependencies/) here instead of hiding the component? Imo it's preferable!

Reply:
![facepalm-crowd](https://github.com/evergreen-ci/spruce/assets/47064971/b239740b-27f9-4f62-9f12-cff59922cc37)

this makes so much more sense